velocity_modelling/cvm/global_model.py

```diff
@@ -0,0 +1,461 @@
+
+    def __init__(
+        self, latitudes: np.ndarray, longitudes: np.ndarray, raster: np.ndarray
+    ):
+        """
+        Initialize the GlobalSurfaceRead.
+        """
+        self.lats = latitudes
+        self.lons = longitudes
+        self.raster = raster
+        self.max_lat = max(self.lats)
+        self.min_lat = min(self.lats)
+        self.max_lon = max(self.lons)
+        self.min_lon = min(self.lons)
+
+    @property
+    def nlat(self) -> int:
+        """
+        Get the number of latitude points.
+
+        Returns
+        -------
+        int
+            Number of latitude points.
+        """
+        return len(self.lats)
+
+    @property
+    def nlon(self) -> int:
+        """
+        Get the number of longitude points.
+
+        Returns
+        -------
+        int
+            Number of longitude points.
+        """
+        return len(self.lons)
+
+
+# TODO:  Under-utilized, perhaps remove?s
+class GlobalSurfaces:
+    """
+    Class to store global surfaces.
+
+    Parameters
+    ----------
+    surfaces : list[GlobalSurfaceRead]
+        List of GlobalSurfaceRead objects.
+
+    Attributes
+    ----------
+    surfaces : list[GlobalSurfaceRead]
+        List of GlobalSurfaceRead objects.
+
+    """
+
+    def __init__(self, surfaces: list[GlobalSurfaceRead]):
+        """
+        Initialize the GlobalSurfaces.
+
+        """
+        self.surfaces = surfaces
+
+
+class PartialGlobalSurfaceDepths:
+    """
+    Class to store partial global surface depths.
+
+    Parameters
+    ----------
+    n_surfaces : int
+        The number of global surfaces.
+
+    Attributes
+    ----------
+    depths : np.ndarray
+        Array of global surface depths.
+
+    """
+
+    def __init__(self, n_surfaces: int):
+        """
+        Initialize the PartialGlobalSurfaceDepths.
+
+        """
+        self.depths = np.zeros(n_surfaces, dtype=np.float64)
+
+    def find_global_submodel_ind_vectorized(self, depths: np.ndarray) -> np.ndarray:
+        """
+        Find the indices of the global sub-velocity model for an array of depths.
+
+        Parameters
+        ----------
+        depths : np.ndarray
+            Array of depths (in m) to find the sub-velocity model indices for.
+
+        Returns
+        -------
+        np.ndarray
+            Array of indices of the global sub-velocity models.
+
+        Raises
+        ------
+        ValueError
+            If any depth is not found in the global sub-velocity model.
+        """
+        depths = np.asarray(depths)
+        max_depth = self.depths[0]
+        valid_mask = depths <= max_depth
+
+        if not np.all(valid_mask):
+            invalid_depths = depths[~valid_mask]
+            raise ValueError(
+                f"Error: Some depths not found in global sub-velocity model: {invalid_depths}"
+            )
+
+        indices = np.searchsorted(self.depths[::-1], depths, side="right")
+        n_velo_indices = len(self.depths) - indices - 1
```

Comment:
Bit of inconsistency here between this and the other `searchsorted` call.

 ```suggestion
        indices = np.searchsorted(-self.depths, -depths, side="right")
```